includes/class-newspack-popups.php

```diff
@@ -1211,5 +1212,33 @@ public static function disable_prompts_for_protected_pages() {
 			}
 		}
 	}
+
+	/**
+	 * If the current session is a preview, ensure that reader data does not persist past the session.
+	 */
+	public static function maybe_create_temp_reader_session() {
+		$is_preview = self::is_preview_request();
+		if ( ! $is_preview ) {
+			return;
+		}
+
+		// For one-off previews, just generate a random number.
+		$session_id = \wp_rand( 0, 9999 );
+
+		// If a view_as session, parse the spec for an ID that can be passed between pageviews in the session.
+		$view_as_spec = Newspack_Popups_View_As::parse_view_as();
+		if ( ! empty( $view_as_spec['session_id'] ) ) {
+			$session_id = $view_as_spec['session_id'];
+		}
+
+		// Tell the reader store to use sessionStorage instead of localStorage, and tie the reader prefix to the session.
+		\add_filter( 'newspack_reader_data_store_is_temp_session', '__return_true' );
+		\add_filter(
+			'newspack_reader_data_store_prefix',
+			function() use ( $session_id ) {
+				return sprintf( 'np_temp_session_%d_', $session_id );
+			}
+		);
+	}
```

Comment:
I wonder if this bit should live under `newspack-plugin` as a core feature of the reader data store. We can use [`is_preview()`](https://developer.wordpress.org/reference/functions/is_preview/) and [`is_customize_preview()`](https://developer.wordpress.org/reference/functions/is_customize_preview/) to always use temporary sessions.

Separate PR, though. We don't have to do this now.

Reply:
I can see that argument! My reasoning for where it is:

- Using the two filters in the main plugin (`newspack_reader_data_store_is_temp_session` and optionally `newspack_reader_data_store_prefix` to create isolated reader data sets) temp sessions can be created for any purpose from any other plugin.
- This particular temp session is created for the specific purpose of previewing Campaigns prompts and segments, so it makes sense in my mind to live in Campaigns.

Reply:
I'm definitely open to revisiting this decision in a future PR, though